### PR TITLE
feat(javascript): support package cache in node workflows

### LINKED ISSUE #1439: Caching dependencies in GitHub Actions
There appears to be no caching of the deps in the GitHub Action Workflows.

Shall this be added? This can greatly speed up runs.

### PR DESCRIPTION
**Pull request by riywo**
This commit adds a new option for NodeProject
to enable package cache option on GitHub Actions workflows.

When this option is enabled, `actions/setup-node` step is configured to use proper `cache` input based on the package manager option.

Close #1439

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/javascript/node-project.ts [hunk 1]
```diff
@@ -163,6 +163,13 @@ export interface NodeProjectOptions
    */
   readonly workflowNodeVersion?: string;
 
+  /**
+   * Enable Node.js package cache in GitHub workflows.
+   *
+   * @default false
+   */
+  readonly workflowNodePackageCacheEnabled?: boolean;
```
Comment (mrgrain) [inline src/javascript/node-project.ts:171]:
```suggestion
  readonly workflowPackageCache?: boolean;
```

**Review by mrgrain — CHANGES_REQUESTED** [review]
Just a small rename

File: src/javascript/node-project.ts (hunk 1 above)
Comment (riywo) [reply src/javascript/node-project.ts]: Renamed!

**Comment by mrgrain** [comment]
@riywo

```
This pull request cannot be embarked for merge
The merge queue pull request can't be updated
Details:

Pull request can't be updated with latest base branch changes

Mergify needs the author permission to update the base branch of the pull request.
https://github.com/opsbr needs to [authorize modification on its head branch](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
err-code: 43C09
```

**Pull request closed without merging**